### PR TITLE
Add logic to start/stop ML recording as Rtc call is started/stopped

### PR DESCRIPTION
**Pull request by gabrysgab**
### Task

 - [JIRA](https://netguru.atlassian.net/browse/BM-533)
 
### Description
Audio source can be used by one process -> implement logic to
start stop audio using processes.
Delete (unused) logic used for sending sound from parent to child.
Delete permissions ask on parent device as it was obsolete.
Format/refactor

### CODE REVIEW

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager.kt [hunk 1]
```diff
@@ -27,8 +30,11 @@ class WebRtcManager constructor(
         cameraEnumerator.deviceNames.asSequence()
             // prefer back-facing cameras
             .sortedBy { deviceName ->
-                if (cameraEnumerator.isBackFacing(deviceName)) 1
-                else 2
+                if (cameraEnumerator.isBackFacing(deviceName)) {
+                    1
+                } else {
+                    2
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager.kt#L36"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerHandler.kt [hunk 2]
```diff
@@ -29,43 +31,49 @@ class WebSocketServerHandler(
...
+                    Timber.e(it)
+                })
+                .addTo(compositeDisposable)
         }
     }
 
-    fun restartServer() {
+    private fun restartServer() {
         stopServer()
         Completable.timer(5, TimeUnit.SECONDS)
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerHandler.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerHandler.kt#L66"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt [hunk 3]
```diff
@@ -21,21 +21,25 @@ class AacRecorder {
 
     fun startRecording(): Completable = Completable.fromAction {
         Timber.i("starting recording")
-        bufferSize = AudioRecord.getMinBufferSize(SAMPLING_RATE, AudioFormat.CHANNEL_IN_MONO, AudioFormat.ENCODING_PCM_16BIT) * 2
+        bufferSize = AudioRecord.getMinBufferSize(
+            SAMPLING_RATE,
+            AudioFormat.CHANNEL_IN_MONO,
+            AudioFormat.ENCODING_PCM_16BIT
+        ) * 2
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt#L28"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt [hunk 4]
```diff
@@ -21,21 +21,25 @@ class AacRecorder {
...
         if (bufferSize == AudioRecord.ERROR || bufferSize == AudioRecord.ERROR_BAD_VALUE) {
             bufferSize = SAMPLING_RATE * 2
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt#L30"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt [hunk 5]
```diff
@@ -73,15 +75,28 @@ class ServerViewModel @Inject constructor(
...
         compositeDisposable.dispose()
         super.onCleared()
     }
+
+    fun handleRtcServerConnectionState(connectionObservable: Observable<RtcServerConnectionState>) {
+        compositeDisposable += connectionObservable.subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeBy(
+                onNext = { mutableRtcConnectionStatus.postValue(it) },
+                onError = { mutableRtcConnectionStatus.postValue(RtcServerConnectionState.Error)}
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt#L95"></span>Missing spacing before “}”</td>
    </tr>
  </tbody>
</table>

**Comment by netguru-hound** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcService.kt#L27">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcService.kt#L27</a> - Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClient.kt#L34">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClient.kt#L34</a> - Expression with labels increase complexity and affect maintainability.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt#L69">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder.kt#L69</a> - This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:link: <a href="https://netguru.atlassian.net/browse/BM-533">BM-533</a></td>
    </tr>
  </tbody>
</table>

### JaCoCO Code Coverage 2.76% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcService` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/machinelearning/AacRecorder` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/machinelearning/MachineLearningService` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/base/RtcCall` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerService` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerHandler` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorViewModel` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ServerViewModel` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClient` | 0% | 0% | :white_check_mark: |


#### APK file size

| size |
| --- |
| 21.6MB |



#### Number of method references

| Dex file | count |
| --- | --- |
| classes.dex | 34373 |



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/RtcServerConnectionState.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+}
+
+sealed class RtcServerConnectionState {
+    object ConnectionOffer : RtcServerConnectionState()
+    object Connected : RtcServerConnectionState()
+    object Disconnected : RtcServerConnectionState()
+    object Error : RtcServerConnectionState()
+}
```
Comment (Aruziell) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/RtcServerConnectionState.kt]: I don't think modeling state this way is a good approach. There was such thing in legacy codebase which caused me headache a couple of times.

Comment (gabrysgab) [reply app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/RtcServerConnectionState.kt]: Any alternatives and why is it bad?

**Pull request merged**